mkosi/kmod.py

```diff
@@ -180,6 +181,14 @@ def gen_required_kernel_modules(
             mods = set(modulesd.rglob("*.ko*"))
         firmware = set()
 
+    # Some firmware dependencies are symbolic links, so the targets for those must be included in the list
+    # of required firmware files too.
+    for fw in firmware.copy():
+        if (root / fw).is_symlink():
+            target = Path(chase(root, fw))
+            if target.exists():
+                firmware.add(target.relative_to(root))
```

Comment:
Maybe extend the comment and mention that we make an implicit assumption here that we only have a single level of symlinks. If we have a symlink to a symlink this approach won't work and we'll end up with a dangling symlink in the initrd.